nucypher/blockchain/eth/actors.py

```diff
@@ -358,10 +347,78 @@ def publish_aggregated_transcript(
             public_key=public_key,
             participant_public_key=participant_public_key,
             transacting_power=self.transacting_power,
-            fire_and_forget=True,
         )
         return tx_hash
 
+    def get_phase_receipt(
+        self, ritual_id: int, phase: int
+    ) -> Tuple[Optional[HexBytes], Optional[TxReceipt]]:
+        if phase == 1:
+            txhash = self.dkg_storage.get_transcript_txhash(ritual_id=ritual_id)
+        elif phase == 2:
+            txhash = self.dkg_storage.get_aggregation_txhash(ritual_id=ritual_id)
+        else:
+            raise ValueError(f"Invalid phase: '{phase}'.")
+        if not txhash:
+            return None, None
+        try:
+            blockchain = self.coordinator_agent.blockchain.client
+            receipt = blockchain.get_transaction_receipt(txhash)
+        except TransactionNotFound:
+            return txhash, None
+        status = receipt.get("status")
+        if status == 1:
+            # If status in response equals 1 the transaction was successful.
+            # If it is equals 0 the transaction was reverted by EVM.
+            # https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.get_transaction_receipt
+            # TODO: What follow-up actions can be taken if the transaction was reverted?
+            return txhash, receipt
+        raise self.ActorError(
```

Comment:
@KPrasch given that we are now going to separate retry/replacement from the act of `perform_round_1`/`perform_round_2`, I don't think (?) we should raise here. 

In the Actor's case, it only cares about whether there is a pending tx or not. In this case, if the original tx was reverted, the tx is not pending (and the  **eventual** dkg tracker (#3424 ) presumably can't do anything about a reverted tx) so should this return `None, None` so that the Actor tries another post transcript/post aggregation tx?

Basically, would raising here potentially cause the node to never get out of a reverted tx situation?

Reply:
We may need to separate `get_phase_receipt` logic to check only for a tx hash (used by Actor), and txhash & receipt (used by DKG Tracker).

Reply:
One potential musing to "fix" this - https://github.com/derekpierre/nucypher/commit/5881403d0a0fbebdf31444be4755b1d8b4e17901. Let me know what you think.

Reply:
lgtm 👍🏻 even with the little amount of direction finding we've done in #3424 it's clear we're not going to fire retry phases in the same async task as `perform_round1|2`.

Reply:
👍 - I'll just add some tests for it.

Reply:
✔️ - done!